pkg/controller/eks/addon/setup.go

```diff
@@ -144,10 +149,69 @@ func (h *hooks) isUpToDate(cr *eksv1alpha1.Addon, resp *awseks.DescribeAddonOutp
 		return false, nil
 	}
 
+	// AddOn Configuration Values
+	configUpToDate, configUpToDateErr := isUpToDateConfigurationValues(cr, resp)
+	if configUpToDateErr != nil {
+		return false, configUpToDateErr
+	}
+	if !configUpToDate {
+		return false, nil
+	}
+
 	add, remove := awsclients.DiffTagsMapPtr(cr.Spec.ForProvider.Tags, resp.Addon.Tags)
 	return len(add) == 0 && len(remove) == 0, nil
 }
 
+func isUpToDateConfigurationValues(cr *eksv1alpha1.Addon, obj *awseks.DescribeAddonOutput) (bool, error) {
+	// Handle nil pointer refs
+	crConfigurationValues := aws.StringValue(cr.Spec.ForProvider.ConfigurationValues)
+	objConfigurationValues := aws.StringValue(obj.Addon.ConfigurationValues)
+	if crConfigurationValues == "" && objConfigurationValues == "" {
+		return true, nil
+	}
+
+	if crConfigurationValues == "" && objConfigurationValues != "" {
+		return false, nil
+	}
+	if crConfigurationValues != "" && objConfigurationValues == "" {
+		return false, nil
+	}
+
+	// Normalize the data
+	objConfigurationValues, objErr := convertConfigurationValuesToString(objConfigurationValues)
+	if objErr != nil {
+		return false, objErr
+	}
+	crConfigurationValues, crErr := convertConfigurationValuesToString(crConfigurationValues)
+	if crErr != nil {
+		return false, crErr
+	}
+	// Compare normalized strings
+	return cmp.Equal(objConfigurationValues, crConfigurationValues), nil
+}
+
+// convertConfigurationValuesToString will deserialize and serialize in order to normalize
+func convertConfigurationValuesToString(configurationValues string) (string, error) {
+	var jsonConfigurationValues map[string]interface{}
+	if strings.HasPrefix(configurationValues, "{") {
+		err := json.Unmarshal([]byte(configurationValues), &jsonConfigurationValues)
+		if err != nil {
+			return configurationValues, err
+		}
+		// Convert back to string for better comparison
+		result, marshalErr := json.Marshal(&jsonConfigurationValues)
+		return string(result), marshalErr
+	}
+	// If YAML
+	err := yaml.Unmarshal([]byte(configurationValues), &jsonConfigurationValues)
+	if err != nil {
+		return configurationValues, err
+	}
+	// Convert back to string for better comparison
+	result, marshalErr := yaml.Marshal(&jsonConfigurationValues)
```

Comment:
I think it would be better to compare the parsed objects instead of the raw strings and use `cmp.Diff` to detect any changes.

Reply:
will update, thanks!